clang/runtime/dpct-rt/include/debug/schema.hpp

```diff
@@ -0,0 +1,391 @@
+                               IsBasicType, ValTy, ValSize, Location));
+}
+
+inline std::shared_ptr<Schema> get_type_schema(dpct_json::value &v) {
+  auto obj = v.get_value<dpct_json::object>();
+  std::string schema_name = obj.get("TypeName").get_value<std::string>();
+
+  std::string schema_type_name =
+      obj.get("schema_type").get_value<std::string>();
+  int field_num = obj.get("FieldNum").get_value<int>();
+  size_t type_size = obj.get("TypeSize").get_value<int>();
+  bool is_virtual = obj.get("IsVirtual").get_value<bool>();
+  std::string file_path = obj.get("FilePath").get_value<std::string>();
+  std::pair<std::string, std::shared_ptr<Schema>> schema_struct_pair =
+      create_schema_struct(schema_name, get_schema_type(schema_type_name),
+                           field_num, is_virtual, type_size, file_path);
+  std::shared_ptr<Schema> struct_schema = schema_struct_pair.second;
+  schema_map[schema_struct_pair.first] = struct_schema;
+
+  std::cout << "name is " << schema_name << "\n type " << schema_type_name
+            << "\n field num " << field_num << "\n is  virtual " << is_virtual
+            << "\n file path is " << file_path << " \ntype size " << type_size
+            << std::endl;
+  if (field_num != 0) {
+    dpct_json::array mem_arr = obj["Members"].get_value<dpct_json::array>();
+    for (int i = 0; i < field_num; i++) {
+      dpct_json::object data_obj = mem_arr[i].get_value<dpct_json::object>();
+      std::string var_name = data_obj.get("VarName").get_value<std::string>();
+      std::string type_name = data_obj.get("TypeName").get_value<std::string>();
+      size_t type_size = obj.get("TypeSize").get_value<int>();
+      bool is_basic_type = data_obj.get("IsBasicType").get_value<bool>();
+      std::string val_type = data_obj.get("ValType").get_value<std::string>();
+      size_t val_size = data_obj.get("ValSize").get_value<int>();
+      size_t offset = data_obj.get("Offset").get_value<int>();
+      std::string location = data_obj.get("Location").get_value<std::string>();
+      create_schema_member(struct_schema, var_name, type_name, type_size,
+                           is_basic_type, getValType(val_type), val_size,
+                           offset, getMemLoc(location));
+    }
+  }
+  return struct_schema;
+}
+
+inline std::shared_ptr<Schema> get_data_schema(dpct_json::value &v) {
+  dpct_json::object data_obj = v.get_value<dpct_json::object>();
+  std::string var_name = data_obj.get("VarName").get_value<std::string>();
+  std::string type_name = data_obj.get("TypeName").get_value<std::string>();
+  const schema_type local_type =
+      get_schema_type(data_obj.get("SchemaType").get_value<std::string>());
+  bool is_basic_type = data_obj.get("IsBasicType").get_value<bool>();
+  size_t type_size = data_obj.get("TypeSize").get_value<int>();
+  std::string val_type = data_obj.get("ValType").get_value<std::string>();
+  size_t val_size = data_obj.get("ValSize").get_value<int>();
```

Comment:
What do TypeSize and ValSize represent？ I think TypeSizeshould be stored in TypeSchema